Bottom: 1ee70d484610b6d5a213777a5b5685b7c78e0302
Top:    92382a2107f25a4c5fc7b8e016bd4c203f9ca80e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 15:21:03 +0000

Update graphics API a little


---

diff --git a/src/crosshair.c b/src/crosshair.c
index 1d20765..5d50d23 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -66,7 +66,7 @@ thindraw_moved_pv (hidGC gc, PinType *pv, Coord x, Coord y)
   moved_pv.X += x;
   moved_pv.Y += y;
 
-  hid_draw_thin_pcb_pv (gc, gc, &moved_pv, true, false);
+  hid_draw__thin_pcb_pv (gc, gc, &moved_pv, true, false);
 }
 
 /* ---------------------------------------------------------------------------
@@ -233,7 +233,7 @@ XORDrawElement (hidGC gc, ElementType *Element, Coord DX, Coord DY)
         moved_pad.Point1.X += DX; moved_pad.Point1.Y += DY;
         moved_pad.Point2.X += DX; moved_pad.Point2.Y += DY;
 
-        hid_draw_thin_pcb_pad (gc, &moved_pad, false, false);
+        hid_draw__thin_pcb_pad (gc, &moved_pad, false, false);
       }
   }
   END_LOOP;
@@ -689,7 +689,7 @@ DrawAttached (hidGC gc)
         via.Mask = 0;
         via.Flags = NoFlags ();
 
-        hid_draw_thin_pcb_pv (gc, gc, &via, true, false);
+        hid_draw__thin_pcb_pv (gc, gc, &via, true, false);
 
         if (TEST_FLAG (SHOWDRCFLAG, PCB))
           {
diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index 5b13d95..39f8d6f 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -10,42 +10,27 @@
 void ghid_set_lock_effects (hidGC gc, AnyObjectType *object);
 
 static void
-_draw_pv (PinType *pv, bool draw_hole)
-{
-  if (TEST_FLAG (THINDRAWFLAG, PCB))
-    hid_draw_thin_pcb_pv (Output.fgGC, Output.fgGC, pv, draw_hole, false);
-  else
-    hid_draw_fill_pcb_pv (Output.fgGC, Output.bgGC, pv, draw_hole, false);
-}
-
-static void
 draw_pin (PinType *pin, const BoxType *drawn_area, void *userdata)
 {
-  _draw_pv (pin, false);
+  hid_draw_pcb_pv (Output.fgGC, Output.bgGC, pin, false, false);
 }
 
 static void
 draw_pin_mask (PinType *pin, const BoxType *drawn_area, void *userdata)
 {
-  if (TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG (THINDRAWPOLYFLAG, PCB))
-    hid_draw_thin_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
-  else
-    hid_draw_fill_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
+  hid_draw_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
 }
 
 static void
 draw_via (PinType *via, const BoxType *drawn_area, void *userdata)
 {
-  _draw_pv (via, false);
+  hid_draw_pcb_pv (Output.fgGC, Output.bgGC, via, false, false);
 }
 
 static void
 draw_via_mask (PinType *via, const BoxType *drawn_area, void *userdata)
 {
-  if (TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG (THINDRAWPOLYFLAG, PCB))
-    hid_draw_thin_pcb_pv (Output.pmGC, Output.pmGC, via, false, true);
-  else
-    hid_draw_fill_pcb_pv (Output.pmGC, Output.pmGC, via, false, true);
+  hid_draw_pcb_pv (Output.pmGC, Output.pmGC, via, false, true);
 }
 
 static void
@@ -69,11 +54,7 @@ _draw_pad (hidGC gc, PadType *pad, bool clear, bool mask)
   if (clear && !mask && pad->Clearance <= 0)
     return;
 
-  if (TEST_FLAG (THINDRAWFLAG, PCB) ||
-      (clear && TEST_FLAG (THINDRAWPOLYFLAG, PCB)))
-    hid_draw_thin_pcb_pad (gc, pad, clear, mask);
-  else
-    hid_draw_fill_pcb_pad (gc, pad, clear, mask);
+  hid_draw_pcb_pad (gc, pad, clear, mask);
 }
 
 static void
diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index 3ada35b..21053b2 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -261,9 +261,9 @@ common_gui_draw_pcb_polygon (hidGC gc, PolygonType *polygon, const BoxType *clip
     return;
 
   if (TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG (THINDRAWPOLYFLAG, PCB))
-    hid_draw_thin_pcb_polygon (gc, polygon, clip_box);
+    hid_draw__thin_pcb_polygon (gc, polygon, clip_box);
   else
-    hid_draw_fill_pcb_polygon (gc, polygon, clip_box);
+    hid_draw__fill_pcb_polygon (gc, polygon, clip_box);
 
   /* If checking planes, thin-draw any pieces which have been clipped away */
   if (TEST_FLAG (CHECKPLANESFLAG, PCB) && !TEST_FLAG (FULLPOLYFLAG, polygon))
@@ -273,7 +273,7 @@ common_gui_draw_pcb_polygon (hidGC gc, PolygonType *polygon, const BoxType *clip
       for (poly.Clipped = polygon->Clipped->f;
            poly.Clipped != polygon->Clipped;
            poly.Clipped = poly.Clipped->f)
-        hid_draw_thin_pcb_polygon (gc, &poly, clip_box);
+        hid_draw__thin_pcb_polygon (gc, &poly, clip_box);
     }
 }
 
@@ -423,6 +423,15 @@ common_thindraw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask)
 }
 
 void
+common_gui_draw_pcb_pad (hidGC gc, PadType *pad, bool clip, bool mask)
+{
+  if (TEST_FLAG (THINDRAWFLAG, PCB))
+    hid_draw__thin_pcb_pad (gc, pad, clip, mask);
+  else
+    hid_draw__fill_pcb_pad (gc, pad, clip, mask);
+}
+
+void
 common_fill_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask)
 {
   Coord w = clear ? (mask ? pad->Mask
@@ -529,6 +538,15 @@ draw_octagon_poly (hidGC gc, Coord X, Coord Y,
 }
 
 void
+common_gui_draw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask)
+{
+  if (TEST_FLAG (THINDRAWFLAG, PCB))
+    hid_draw__thin_pcb_pv (fg_gc, fg_gc, pv, drawHole, mask); /* All thindraw callers used the same GC for fg and bg */
+  else
+    hid_draw__fill_pcb_pv (fg_gc, bg_gc, pv, drawHole, mask);
+}
+
+void
 common_fill_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask)
 {
   Coord w = mask ? pv->Mask : pv->Thickness;
@@ -626,15 +644,17 @@ common_thindraw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bo
 void
 common_draw_helpers_init (HID_DRAW *graphics)
 {
-  graphics->draw_pcb_line        = common_draw_pcb_line;
-  graphics->draw_pcb_arc         = common_draw_pcb_arc;
-  graphics->draw_pcb_text        = common_draw_pcb_text;
-  graphics->draw_pcb_polygon     = common_fill_pcb_polygon; /* Default is the non-GUI case */
-
-  graphics->fill_pcb_polygon     = common_fill_pcb_polygon;
-  graphics->thindraw_pcb_polygon = common_thindraw_pcb_polygon;
-  graphics->fill_pcb_pad         = common_fill_pcb_pad;
-  graphics->thindraw_pcb_pad     = common_thindraw_pcb_pad;
-  graphics->fill_pcb_pv          = common_fill_pcb_pv;
-  graphics->thindraw_pcb_pv      = common_thindraw_pcb_pv;
+  graphics->draw_pcb_line         = common_draw_pcb_line;
+  graphics->draw_pcb_arc          = common_draw_pcb_arc;
+  graphics->draw_pcb_text         = common_draw_pcb_text;
+  graphics->draw_pcb_polygon      = common_fill_pcb_polygon; /* Default is the non-GUI case */
+  graphics->draw_pcb_pad          = common_fill_pcb_pad;     /* Default is the non-GUI case */
+  graphics->draw_pcb_pv           = common_fill_pcb_pv;      /* Default is the non-GUI case */
+
+  graphics->_fill_pcb_polygon     = common_fill_pcb_polygon;
+  graphics->_thindraw_pcb_polygon = common_thindraw_pcb_polygon;
+  graphics->_fill_pcb_pad         = common_fill_pcb_pad;
+  graphics->_thindraw_pcb_pad     = common_thindraw_pcb_pad;
+  graphics->_fill_pcb_pv          = common_fill_pcb_pv;
+  graphics->_thindraw_pcb_pv      = common_thindraw_pcb_pv;
 }
diff --git a/src/hid/common/draw_helpers.h b/src/hid/common/draw_helpers.h
index fb5730c..eb95879 100644
--- a/src/hid/common/draw_helpers.h
+++ b/src/hid/common/draw_helpers.h
@@ -1,8 +1,10 @@
 void common_gui_draw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box);
 void common_fill_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box);
 void common_thindraw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box);
+void common_gui_draw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask);
 void common_fill_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask);
 void common_thindraw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask);
+void common_gui_draw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
 void common_fill_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
 void common_thindraw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
 void common_draw_helpers_init (HID_DRAW *graphics);
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index ed45bad..04246eb 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -149,25 +149,13 @@ nogui_draw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
 }
 
 static void
-nogui_fill_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask)
+nogui_draw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask)
 {
   CRASH;
 }
 
 static void
-nogui_thindraw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask)
-{
-  CRASH;
-}
-
-static void
-nogui_fill_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pad, bool drawHole, bool mask)
-{
-  CRASH;
-}
-
-static void
-nogui_thindraw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pad, bool drawHole, bool mask)
+nogui_draw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pad, bool drawHole, bool mask)
 {
   CRASH;
 }
@@ -497,10 +485,8 @@ common_nogui_graphics_init (HID_DRAW *graphics)
   graphics->fill_rect =       nogui_fill_rect;
 
   graphics->draw_pcb_polygon = nogui_draw_pcb_polygon;
-  graphics->fill_pcb_pad =     nogui_fill_pcb_pad;
-  graphics->thindraw_pcb_pad = nogui_thindraw_pcb_pad;
-  graphics->fill_pcb_pv =      nogui_fill_pcb_pv;
-  graphics->thindraw_pcb_pv =  nogui_thindraw_pcb_pv;
+  graphics->draw_pcb_pad     = nogui_draw_pcb_pad;
+  graphics->draw_pcb_pv      = nogui_draw_pcb_pv;
 }
 
 static HID nogui_hid;
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 30202e4..9d4f2c0 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -899,7 +899,7 @@ ghid_thindraw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
   double old_alpha_mult = gtk_gc->alpha_mult;
   common_thindraw_pcb_polygon (gc, poly, clip_box);
   ghid_set_alpha_mult (gc, gtk_gc->alpha_mult * 0.25);
-  hid_draw_fill_pcb_polygon (gc, poly, clip_box);
+  hid_draw__fill_pcb_polygon (gc, poly, clip_box);
   ghid_set_alpha_mult (gc, old_alpha_mult);
 }
 
@@ -1109,8 +1109,8 @@ ghid_init_renderer (int *argc, char ***argv, GHidPort *port)
 
   /* Setup HID function pointers specific to the GL renderer*/
   ghid_hid.end_layer = ghid_end_layer;
-  ghid_graphics.fill_pcb_polygon = ghid_fill_pcb_polygon;
-  ghid_graphics.thindraw_pcb_polygon = ghid_thindraw_pcb_polygon;
+  ghid_graphics._fill_pcb_polygon = ghid_fill_pcb_polygon;
+  ghid_graphics._thindraw_pcb_polygon = ghid_thindraw_pcb_polygon;
 }
 
 void
@@ -1288,9 +1288,9 @@ static void
 _draw_pv (PinType *pv, bool draw_hole)
 {
   if (TEST_FLAG (THINDRAWFLAG, PCB))
-    hid_draw_thin_pcb_pv (Output.fgGC, Output.fgGC, pv, draw_hole, false);
+    hid_draw__thin_pcb_pv (Output.fgGC, Output.fgGC, pv, draw_hole, false);
   else
-    hid_draw_fill_pcb_pv (Output.fgGC, Output.bgGC, pv, draw_hole, false);
+    hid_draw__fill_pcb_pv (Output.fgGC, Output.bgGC, pv, draw_hole, false);
 
   if (!TEST_FLAG (HOLEFLAG, pv) && TEST_FLAG (DISPLAYNAMEFLAG, pv))
     _draw_pv_name (pv);
@@ -1409,9 +1409,9 @@ _draw_pad (hidGC gc, PadType *pad, bool clear, bool mask)
 
   if (TEST_FLAG (THINDRAWFLAG, PCB) ||
       (clear && TEST_FLAG (THINDRAWPOLYFLAG, PCB)))
-    hid_draw_thin_pcb_pad (gc, pad, clear, mask);
+    hid_draw__thin_pcb_pad (gc, pad, clear, mask);
   else
-    hid_draw_fill_pcb_pad (gc, pad, clear, mask);
+    hid_draw__fill_pcb_pad (gc, pad, clear, mask);
 }
 
 static void
@@ -1577,9 +1577,9 @@ clearPin_callback (const BoxType * b, void *cl)
 {
   PinType *pin = (PinType *) b;
   if (TEST_FLAG (THINDRAWFLAG, PCB) || TEST_FLAG (THINDRAWPOLYFLAG, PCB))
-    hid_draw_thin_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
+    hid_draw__thin_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
   else
-    hid_draw_fill_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
+    hid_draw__fill_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
   return 1;
 }
 
@@ -1597,7 +1597,7 @@ static int
 clearPin_callback_solid (const BoxType * b, void *cl)
 {
   PinType *pin = (PinType *) b;
-  hid_draw_fill_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
+  hid_draw__fill_pcb_pv (Output.pmGC, Output.pmGC, pin, false, true);
   return 1;
 }
 
@@ -1607,7 +1607,7 @@ clearPad_callback_solid (const BoxType * b, void *cl)
   PadType *pad = (PadType *) b;
   int *side = cl;
   if (ON_SIDE (pad, *side) && pad->Mask)
-    hid_draw_fill_pcb_pad (Output.pmGC, pad, true, true);
+    hid_draw__fill_pcb_pad (Output.pmGC, pad, true, true);
   return 1;
 }
 
@@ -1637,7 +1637,7 @@ fill_board_outline (hidGC gc, const BoxType *drawn_area)
     polygon.BoundingBox = *drawn_area;
   polygon.Flags = NoFlags ();
   SET_FLAG (FULLPOLYFLAG, &polygon);
-  hid_draw_fill_pcb_polygon (gc, &polygon, drawn_area);
+  hid_draw__fill_pcb_polygon (gc, &polygon, drawn_area);
   poly_FreeContours (&polygon.NoHoles);
 }
 
@@ -1669,7 +1669,7 @@ fill_outline_hole_cb (PLINE *pl, void *user_data)
 
   /* XXX: For some reason, common_fill_pcb_polygon doesn't work for all contours here.. not sure why */
 //  common_fill_pcb_polygon (info->gc, &polygon, NULL);
-  hid_draw_fill_pcb_polygon (info->gc, &polygon, NULL);
+  hid_draw__fill_pcb_polygon (info->gc, &polygon, NULL);
 
   poly_FreeContours (&polygon.NoHoles);
 
@@ -1787,7 +1787,7 @@ GhidDrawMask (int side, BoxType * screen)
     polygon.BoundingBox = *screen;
   polygon.Flags = NoFlags ();
   SET_FLAG (FULLPOLYFLAG, &polygon);
-  hid_draw_fill_pcb_polygon (out->fgGC, &polygon, screen);
+  hid_draw__fill_pcb_polygon (out->fgGC, &polygon, screen);
   poly_FreeContours (&polygon.NoHoles);
 #endif
 
@@ -2345,7 +2345,7 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
      we can't use the hidgl polygon drawing routine */
   /* TODO: We could use the GLU tessellator though */
   if (hidgl_stencil_bits (priv->hidgl) == 0)
-    ghid_graphics.fill_pcb_polygon = common_fill_pcb_polygon;
+    ghid_graphics._fill_pcb_polygon = common_fill_pcb_polygon;
 
   glEnable (GL_BLEND);
   glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index c68b9d3..38dee48 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2212,6 +2212,8 @@ hid_gtk_init ()
   ghid_graphics.fill_rect           = ghid_fill_rect;
 
   ghid_graphics.draw_pcb_polygon    = common_gui_draw_pcb_polygon;
+  ghid_graphics.draw_pcb_pad        = common_gui_draw_pcb_pad;
+  ghid_graphics.draw_pcb_pv         = common_gui_draw_pcb_pv;
 
   hid_register_hid (&ghid_hid);
 #include "gtk_lists.h"
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index ca524c7..9408227 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -4129,6 +4129,8 @@ hid_lesstif_init ()
   lesstif_graphics.fill_rect           = lesstif_fill_rect;
 
   lesstif_graphics.draw_pcb_polygon    = common_gui_draw_pcb_polygon;
+  lesstif_graphics.draw_pcb_pad        = common_gui_draw_pcb_pad;
+  lesstif_graphics.draw_pcb_pv         = common_gui_draw_pcb_pv;
 
   hid_register_hid (&lesstif_hid);
 #include "lesstif_lists.h"
diff --git a/src/hid_draw.h b/src/hid_draw.h
index ad984f9..276d04b 100644
--- a/src/hid_draw.h
+++ b/src/hid_draw.h
@@ -55,13 +55,16 @@ struct hid_draw_st
   void (*draw_pcb_arc) (hidGC gc, ArcType *arc);
   void (*draw_pcb_text) (hidGC gc, TextType *, Coord);
   void (*draw_pcb_polygon) (hidGC gc, PolygonType *poly, const BoxType *clip_box);
+  void (*draw_pcb_pad) (hidGC gc, PadType *pad, bool clip, bool mask);
+  void (*draw_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
 
-  void (*fill_pcb_polygon) (hidGC gc, PolygonType *poly, const BoxType *clip_box);
-  void (*thindraw_pcb_polygon) (hidGC gc, PolygonType *poly, const BoxType *clip_box);
-  void (*fill_pcb_pad) (hidGC gc, PadType *pad, bool clip, bool mask);
-  void (*thindraw_pcb_pad) (hidGC gc, PadType *pad, bool clip, bool mask);
-  void (*fill_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
-  void (*thindraw_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
+  /* The following are not meant to be called outside of the GUI implementations of the above APIs */
+  void (*_fill_pcb_polygon) (hidGC gc, PolygonType *poly, const BoxType *clip_box);
+  void (*_thindraw_pcb_polygon) (hidGC gc, PolygonType *poly, const BoxType *clip_box);
+  void (*_fill_pcb_pad) (hidGC gc, PadType *pad, bool clip, bool mask);
+  void (*_thindraw_pcb_pad) (hidGC gc, PadType *pad, bool clip, bool mask);
+  void (*_fill_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
+  void (*_thindraw_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
 
 };
 
@@ -182,39 +185,51 @@ hid_draw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
   gc->hid_draw->draw_pcb_polygon (gc, poly, clip_box);
 }
 
+inline void
+hid_draw_pcb_pad (hidGC gc, PadType *pad, bool clip, bool mask)
+{
+  gc->hid_draw->draw_pcb_pad (gc, pad, clip, mask);
+}
+
+inline void
+hid_draw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool draw_hole, bool mask)
+{
+  fg_gc->hid_draw->draw_pcb_pv (fg_gc, bg_gc, pv, draw_hole, mask);
+}
+
 
 inline void
-hid_draw_fill_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
+hid_draw__fill_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
 {
-  gc->hid_draw->fill_pcb_polygon (gc, poly, clip_box);
+  gc->hid_draw->_fill_pcb_polygon (gc, poly, clip_box);
 }
 
 inline void
-hid_draw_thin_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
+hid_draw__thin_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
 {
-  gc->hid_draw->thindraw_pcb_polygon (gc, poly, clip_box);
+  gc->hid_draw->_thindraw_pcb_polygon (gc, poly, clip_box);
 }
 
 inline void
-hid_draw_fill_pcb_pad (hidGC gc, PadType *pad, bool clip, bool mask)
+hid_draw__fill_pcb_pad (hidGC gc, PadType *pad, bool clip, bool mask)
 {
-  gc->hid_draw->fill_pcb_pad (gc, pad, clip, mask);
+  gc->hid_draw->_fill_pcb_pad (gc, pad, clip, mask);
 }
 
 inline void
-hid_draw_thin_pcb_pad (hidGC gc, PadType *pad, bool clip, bool mask)
+hid_draw__thin_pcb_pad (hidGC gc, PadType *pad, bool clip, bool mask)
 {
-  gc->hid_draw->thindraw_pcb_pad (gc, pad, clip, mask);
+  gc->hid_draw->_thindraw_pcb_pad (gc, pad, clip, mask);
 }
 
 inline void
-hid_draw_fill_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool draw_hole, bool mask)
+hid_draw__fill_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool draw_hole, bool mask)
 {
-  fg_gc->hid_draw->fill_pcb_pv (fg_gc, bg_gc, pv, draw_hole, mask);
+  fg_gc->hid_draw->_fill_pcb_pv (fg_gc, bg_gc, pv, draw_hole, mask);
 }
 
 inline void
-hid_draw_thin_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool draw_hole, bool mask)
+hid_draw__thin_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool draw_hole, bool mask)
 {
-  fg_gc->hid_draw->thindraw_pcb_pv (fg_gc, bg_gc, pv, draw_hole, mask);
+  fg_gc->hid_draw->_thindraw_pcb_pv (fg_gc, bg_gc, pv, draw_hole, mask);
 }
